File: Matlab/occlusionMeasure.m
function [occMeasure] = occlusionMeasure(normIrisMaskImage)
%occlusionMeasure Function that computes the occlusion measure, based on
%the amount of occlusion of the segmented iris

% Most of the variables names are following the paper Feature correlation evaluation approach for iris feature
% quality measure (by Craig Belcher, Yingzi Du, Zhi Zhou and Robert Ives) nomenclature.

%% Variables initialization
[rows, cols] = size(normIrisMaskImage);
kappa = 0.9179;
lambda = 2.5;
segMaskSize = rows*cols;

%% Computing the measure
% black pixels are occlusions, and white pixels are valid iris area
invalidArea = sum(sum(normIrisMaskImage == 0));
O = invalidArea/segMaskSize;

%% Computing the measure normalization
occMeasure = (1 - exp(-lambda*(1 - O)))/kappa;

end

File: FeatureCorrelation/occlusion_measure.py
import numpy as np
from math import exp

# Function that computes a iris segmentation occlusion quality measure
# The normalized mask image is a binary image. White pixels are valid iris area and black invalid
# Input: norm_iris_mask_image (numpy array)
# Output: (Float)
def occlusion_measure(norm_iris_mask_image):
    # Parameters for normalize, according to the reference paper (MORE INFO IN THE MAIN FUNCTION)
    kappa = 0.9179
    lmbda = 2.5

    # total amount of pixels in the image
    img_n_pixels = float(norm_iris_mask_image.size)

    # Computing the occlusion, which is the relation between the area of invalid 
    # pixels (occlusion of the iris) witch the total amount of pixels in the image
    invalid_area = sum(sum(norm_iris_mask_image == 0))
    o = invalid_area/img_n_pixels

    # Computing the quality measure normalization
    return (1 - exp(-lmbda*(1 - o)))/kappa
